compiler/rustc_const_eval/src/interpret/intrinsics/simd.rs

```diff
@@ -0,0 +1,785 @@
+            | sym::simd_div
+            | sym::simd_rem
+            | sym::simd_shl
+            | sym::simd_shr
+            | sym::simd_and
+            | sym::simd_or
+            | sym::simd_xor
+            | sym::simd_eq
+            | sym::simd_ne
+            | sym::simd_lt
+            | sym::simd_le
+            | sym::simd_gt
+            | sym::simd_ge
+            | sym::simd_fmax
+            | sym::simd_fmin
+            | sym::simd_saturating_add
+            | sym::simd_saturating_sub
+            | sym::simd_arith_offset => {
+                use mir::BinOp;
+
+                let (left, left_len) = self.project_to_simd(&args[0])?;
+                let (right, right_len) = self.project_to_simd(&args[1])?;
+                let (dest, dest_len) = self.project_to_simd(dest)?;
+
+                assert_eq!(dest_len, left_len);
+                assert_eq!(dest_len, right_len);
+
+                enum Op {
+                    MirOp(BinOp),
+                    SaturatingOp(BinOp),
+                    FMinMax(MinMax),
+                    WrappingOffset,
+                }
+                let which = match intrinsic_name {
+                    sym::simd_add => Op::MirOp(BinOp::Add),
+                    sym::simd_sub => Op::MirOp(BinOp::Sub),
+                    sym::simd_mul => Op::MirOp(BinOp::Mul),
+                    sym::simd_div => Op::MirOp(BinOp::Div),
+                    sym::simd_rem => Op::MirOp(BinOp::Rem),
+                    sym::simd_shl => Op::MirOp(BinOp::ShlUnchecked),
+                    sym::simd_shr => Op::MirOp(BinOp::ShrUnchecked),
+                    sym::simd_and => Op::MirOp(BinOp::BitAnd),
+                    sym::simd_or => Op::MirOp(BinOp::BitOr),
+                    sym::simd_xor => Op::MirOp(BinOp::BitXor),
+                    sym::simd_eq => Op::MirOp(BinOp::Eq),
+                    sym::simd_ne => Op::MirOp(BinOp::Ne),
+                    sym::simd_lt => Op::MirOp(BinOp::Lt),
+                    sym::simd_le => Op::MirOp(BinOp::Le),
+                    sym::simd_gt => Op::MirOp(BinOp::Gt),
+                    sym::simd_ge => Op::MirOp(BinOp::Ge),
+                    sym::simd_fmax => Op::FMinMax(MinMax::Max),
+                    sym::simd_fmin => Op::FMinMax(MinMax::Min),
+                    sym::simd_saturating_add => Op::SaturatingOp(BinOp::Add),
+                    sym::simd_saturating_sub => Op::SaturatingOp(BinOp::Sub),
+                    sym::simd_arith_offset => Op::WrappingOffset,
+                    _ => unreachable!(),
+                };
+
+                for i in 0..dest_len {
+                    let left = self.read_immediate(&self.project_index(&left, i)?)?;
+                    let right = self.read_immediate(&self.project_index(&right, i)?)?;
+                    let dest = self.project_index(&dest, i)?;
+                    let val = match which {
+                        Op::MirOp(mir_op) => {
+                            // self does NaN adjustments.
```

Comment:
```suggestion
                            // This does NaN adjustments.
```
Please double-check this for the rest of the file.